src/azure-cli/azure/cli/command_modules/acr/tests/latest/test_acr_commands_mock.py

```diff
@@ -387,6 +399,426 @@ def test_repository_delete(self, mock_requests_delete, mock_get_manifest_digest,
+    @mock.patch('azure.cli.command_modules.acr.repository._get_manifest_digest', autospec=True)
+    @mock.patch('requests.request', autospec=True)
+    def test_manifest_metadata_update(self, mock_requests_patch, mock_get_manifest_digest, mock_get_access_credentials):
+        cmd = self._setup_cmd()
+
+        response = mock.MagicMock()
+        response.headers = {}
+        response.status_code = 200
+        response.content = json.dumps({
+            'registry': 'testregistry.azurecr.io',
+            'imageName': 'testrepository'
+        }).encode()
+
+        mock_requests_patch.return_value = response
+        mock_get_access_credentials.return_value = 'testregistry.azurecr.io', 'username', 'password'
+        mock_get_manifest_digest.return_value = 'sha256:c5515758d4c5e1e838e9cd307f6c6a0d620b5e07e6f927b07d05f6d12a1ac8d7'
+
+        # Update attributes for an artifact by tag
+        acr_manifest_metadata_update(cmd,
+                                     registry_name='testregistry',
+                                     manifest_spec='testrepository:testtag',
+                                     write_enabled='false')
+        mock_requests_patch.assert_called_with(
+            method='patch',
+            url='https://testregistry.azurecr.io/acr/v1/testrepository/_tags/testtag',
+            headers=get_authorization_header('username', 'password'),
+            params=None,
+            json={
+                'writeEnabled': 'false'
+            },
+            timeout=300,
+            verify=mock.ANY)
+
+        # Update attributes for an artifact by manifest digest
+        acr_manifest_metadata_update(cmd,
+                                     registry_name='testregistry',
+                                     manifest_spec='testrepository@sha256:c5515758d4c5e1e838e9cd307f6c6a0d620b5e07e6f927b07d05f6d12a1ac8d7',
+                                     write_enabled='false')
+        mock_requests_patch.assert_called_with(
+            method='patch',
+            url='https://testregistry.azurecr.io/acr/v1/testrepository/_manifests/sha256:c5515758d4c5e1e838e9cd307f6c6a0d620b5e07e6f927b07d05f6d12a1ac8d7',
+            headers=get_authorization_header('username', 'password'),
+            params=None,
+            json={
+                'writeEnabled': 'false'
+            },
+            timeout=300,
+            verify=mock.ANY)
+
+        # Update attributes for an artifact by fqdn
+        acr_manifest_metadata_update(cmd,
+                                     manifest_id=['testregistry.azurecr.io/testrepository:testtag'],
+                                     write_enabled='false')
+        mock_requests_patch.assert_called_with(
+            method='patch',
+            url='https://testregistry.azurecr.io/acr/v1/testrepository/_tags/testtag',
+            headers=get_authorization_header('username', 'password'),
+            params=None,
+            json={
```

Comment:
Thanks for adding all the necessary tests. Please make sure your test touches all the new code. At the very least all the API calls.